apps/HelloiOS/HelloiOS/HalideView.mm

```diff
@@ -0,0 +1,364 @@
+            CGSize drawableSize = self.bounds.size;
+
+            // The Metal schedule for our Halide code requires that 
+            // the image be exact multiples of 8 in x & y
+            drawableSize.width  = ((long)drawableSize.width + 7) & ~7;
+            drawableSize.height  = ((long)drawableSize.height + 7) & ~7;
+            _metalLayer.drawableSize = drawableSize;
+            
+            [self initBufsWithWidth: drawableSize.width height: drawableSize.height using_metal: using_metal];
+            halide_funcs.init((__bridge void *)self, buf1);
+
+            [self resetFrameTime];
+        }
+        
+        [self renderOneFrame: halide_funcs using_metal: using_metal];
+
+        id <MTLBuffer> buffer = using_metal ?
+            (__bridge id <MTLBuffer>)(void *)halide_metal_get_buffer((void *)&self, pixel_buf) :
+            [self.device newBufferWithBytes: pixel_buf.data()
+                                     length: pixel_buf.size_in_bytes()
+                                    options:MTLResourceStorageModeShared];
+        id <MTLCommandBuffer> commandBuffer = [_commandQueue commandBuffer];
+        id <MTLBlitCommandEncoder> blitEncoder = [commandBuffer blitCommandEncoder];
+
+        MTLSize image_size;
+        image_size.width = pixel_buf.dim(0).extent();
+        image_size.height = pixel_buf.dim(1).extent();
+        image_size.depth = 1;
+        MTLOrigin origin = { 0, 0, 0 };
+
+        const int bytesPerRow = pixel_buf.dim(1).stride() * pixel_buf.type().bits / 8;
+        [blitEncoder 
+            copyFromBuffer:buffer 
+            sourceOffset: 0
+            sourceBytesPerRow: bytesPerRow
+            sourceBytesPerImage: pixel_buf.size_in_bytes()
+            sourceSize: image_size 
+            toTexture: drawable.texture
+            destinationSlice: 0 
+            destinationLevel: 0 
+            destinationOrigin: origin];
+        [blitEncoder endEncoding];
+        [commandBuffer addCompletedHandler: ^(id MTLCommandBuffer) {
+            dispatch_async(dispatch_get_main_queue(), ^(void) {
+                [drawable present];
+                [self initiateRender];
+            });
+        }];
+        [commandBuffer commit];
+        [_commandQueue insertDebugCaptureBoundary];
+    }
+#else
+    float f = self.contentScaleFactor;
+    int image_width = (int) (self.bounds.size.width * f);
+    int image_height = (int) (self.bounds.size.height * f);
+    const HalideFuncs &halide_funcs = kHalideCPU;
+
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        [self initBufsWithWidth:image_width height:image_height using_metal: false];
+
+        CGDataProviderRef provider =
+            CGDataProviderCreateWithData(NULL, pixel_buf.data(), pixel_buf.size_in_bytes(), NULL);
+        CGColorSpaceRef color_space = CGColorSpaceCreateDeviceRGB();
+
+        halide_funcs.init((__bridge void *)self, buf1);
+        [self resetFrameTime];
+   
+        for (;;) {
```

Comment:
Mimicking the existing approach from previous HelloiOS. Definitely open for better suggestions.